### PR TITLE
Add support for QMP6988 Pressure sensor

### PR DESCRIPTION
**Pull request by andrewpc**
# What does this implement/fix? 

This change adds support for the QMP6988 Pressure and Temperature sensor. This sensor is found in the M5Stack ENVIII alongside the SHT30 (which is already supported in esphome).

Datasheet can be found here: https://m5stack.oss-cn-shenzhen.aliyuncs.com/resource/docs/datasheet/unit/enviii/QMP6988%20Datasheet.pdf

Code reworked to be esphome native from arduino example provided in the following github repo: https://github.com/m5stack/UNIT_ENV 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

## Test Environment

- [x] ESP32
- [x] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
- platform: qmp6988
    temperature:
      name: "Living Room Temperature"
      oversampling: 16x
    pressure:
      name: "Living Room Pressure"
      oversampling: 16x
    address: 0x70
    update_interval: 60s
    iir_filter: 2x
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).
     https://github.com/esphome/esphome-docs/pull/1935

### CODE REVIEW

**Comment by randybb** [comment]
Nice!
```
[13:56:36][D][sensor:125]: 'DEV 1 SHT30 Temperature': Sending state 22.35638 °C with 1 decimals of accuracy
[13:56:36][D][sensor:125]: 'DEV 1 SHT30 Humidity': Sending state 37.43801 % with 1 decimals of accuracy
[13:57:18][D][sensor:125]: 'DEV 1 BME280 Temperature': Sending state 21.68000 °C with 1 decimals of accuracy
[13:57:18][D][sensor:125]: 'DEV 1 BME280 Pressure': Sending state 1010.89648 hPa with 1 decimals of accuracy
[13:57:18][D][sensor:125]: 'DEV 1 BME280 Humidity': Sending state 37.98145 % with 1 decimals of accuracy
[14:02:16][D][sensor:125]: 'DEV 1 QMP6988 Temperature': Sending state 22.53906 °C with 1 decimals of accuracy
[14:02:16][D][sensor:125]: 'DEV 1 QMP6988 Pressure': Sending state 1009.96124 hPa with 1 decimals of accuracy
```

File: esphome/components/qmp6988/qmp6988.cpp [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+  return 1;
+}
+
+uint8_t QMP6988Component::read_data_(uint8_t reg_add, unsigned char *read, uint8_t num) {
+  if (this->read_register(reg_add, read, num) != i2c::ERROR_OK) {
+    return 0;
+  }
+
+  return 1;
+}
```
Comment (jesserockz) [inline esphome/components/qmp6988/qmp6988.cpp]: Why create functions that do not add any extra value? Just use the `write_byte` and `read_register` in the code and check for `!= i2c::ERROR_OK`.
It makes it more confusing this way, espeically when you basically flipped the return value `i2c::ERROR_OK` is `0`

File: esphome/components/qmp6988/qmp6988.cpp [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+
+uint8_t QMP6988Component::device_check_() {
```
Comment (jesserockz) [inline esphome/components/qmp6988/qmp6988.cpp]: This could easily be a `bool` return value

File: esphome/components/qmp6988/qmp6988.cpp [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+  }
+  ESP_LOGD(TAG, "qmp6988 read chip id = 0x%x", qmp6988_data_.chip_id);
+  if (qmp6988_data_.chip_id == QMP6988_CHIP_ID) {
+    return 1;
+  }
+
+  return 0;
+}
+
+int QMP6988Component::get_calibration_data_() {
```
Comment (jesserockz) [inline esphome/components/qmp6988/qmp6988.cpp]: This could be `bool` too

File: esphome/components/qmp6988/qmp6988.cpp [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+  ret = this->write_register_(QMP6988_RESET_REG, 0xe6);
+  if (ret == 0) {
+    ESP_LOGE(TAG, "Software Reset (0xe6) failed");
+  }
+  delay(10);
+
+  this->write_register_(QMP6988_RESET_REG, 0x00);
+}
+
+void QMP6988Component::set_power_mode_(int power_mode) {
```
Comment (jesserockz) [inline esphome/components/qmp6988/qmp6988.cpp]:
```suggestion
void QMP6988Component::set_power_mode_(uint8_t power_mode) {
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/qmp6988/qmp6988.cpp (hunk 1 above)
Comment (andrewpc) [reply esphome/components/qmp6988/qmp6988.cpp]: Ok, functions removed, thanks

**Comment by andrewpc** [comment]
@jesserockz Any update on this one? All the changes you requested should now be there. Thanks

**Review by jesserockz — APPROVED** [review]

**Pull request merged**